pkgs/tools/networking/haproxy/default.nix

```diff
@@ -1,4 +1,8 @@
-{ stdenv, pkgs, fetchurl, openssl, zlib }:
+{ useLua ? false
```

Comment:
Nothing will rebuild if `pcre` is not used in the derivation when `usePcre` is disabled. Adding a default for the optional packages is not strictly necessary but it makes it easier to import the file directly, another thing that's usually added in this case is an assert to make sure the package exists when the use option is enabled.

```nix
{ stdenv, openssl, zlib
, usePcre ? true, pcre ? null
, useLua ? !stdenv.isDarwin, lua ? null
}:

assert usePcre -> pcre != null;
assert useLua -> lua != null;
# ...
```